libchisel/src/remapstart.rs

```diff
@@ -0,0 +1,181 @@
+use std::collections::HashMap;
+
+use super::{ModuleError, ModulePreset, ModuleTranslator};
+use parity_wasm::elements::*;
+
+pub struct RemapStart {}
+
+impl ModulePreset for RemapStart {
+    fn with_preset(preset: &str) -> Result<Self, ()> {
+        match preset {
+            "ewasm" => Ok(RemapStart {}),
+            _ => Err(()),
+        }
+    }
+}
+
+impl ModuleTranslator for RemapStart {
+    fn translate_inplace(&self, module: &mut Module) -> Result<bool, ModuleError> {
+        Ok(remap_start(module))
+    }
+
+    fn translate(&self, module: &Module) -> Result<Option<Module>, ModuleError> {
+        let mut ret = module.clone();
+        let modified = remap_start(&mut ret);
+        if modified {
+            return Ok(Some(ret));
+        }
+        Ok(None)
+    }
+}
+
+// if the main export doesn't refer to the same function as the start section,
+// point the main export towards the function referenced by the start section
+// remove the start section
+fn replace_export(module: &mut Module, export_name: &str, replacement_func_idx: u32) -> bool {
+    let mut ret = false;
+
+    /*
+    let main_func_export_idx = module
+        .export_section()
+        .expect("export section should exist")
+        .entries()
+        .iter()
+        .find_map(|s| match s.internal() {
+            Internal::Function(f) => {
+                if s.field() == "main".to_string() {
+                    Some(*f)
+                } else {
+                    None
+                }
+            }
+            _ => None,
+        })
+        .expect("main export not found"); // TODO can I expect that the module is valid ewasm at this point?
+        */
+
+    let start_func_idx = module.start_section().unwrap();
```

Comment:
This can panic if the module doesn't have a start section. Please include `expect`, if not a `match` to handle the cases explicitly.

I know this is only called after the caller `remap_start` asserts that the start section is `Some`, but still harder to understand if the checking is done somewhere else in the code implicitly.